dts/bindings/sensor/st,stm32-qdec.yaml

```diff
@@ -16,6 +16,34 @@ properties:
   pinctrl-names:
     required: true
 
+  st,encoder-mode:
+    type: int
+    description: |
+      Set encoder mode.
+      The following modes are supported:
+      0x1: Encoder mode 1
+      0x2: Encoder mode 2
+      0x3: Encoder mode 3
+      Some MCUs supports additional modes:
+      0x10002: Encoder mode: Clock plus direction, x2 mode
+      0x10003: Encoder mode: Clock plus direction, x1 mode
+      0x10004: Encoder mode: Directional Clock, x2 mode
+      0x10005: Encoder mode: Directional Clock, x1 mode
+      0x10006: Quadrature encoder mode: x1 mode, counting on tim_ti1fp1
+      0x10007: Quadrature encoder mode: x1 mode, counting on tim_ti2fp2
+
+      If unset encoder mode is defined by LL_TIM_ENCODER_StructInit().
```

Comment:
Why not using a default value which will be matching LL default? See https://docs.zephyrproject.org/latest/build/dts/bindings-syntax.html#default

Also, bindings should not reference code (they are supposed to be 'implementation agnostic' ) but refer to behavior. So you should rather document "what will be the effect when not set".

Note, if there are 2 types of encoding IPs which have different behavior which makes it complex to document with a single description, instantiate a second compatible where your can document a specific behavior.

Reply:
Good question.
I went through the HAL and all have the same default value.
I've added default which shouldn't break backwards compatibility.

Thanks for the additional bindings insight :+1: